src/battle-dex.ts

```diff
@@ -484,7 +484,7 @@ const Dex = new class implements ModdedDex {
 			window.BattlePokedex[id] = species;
 		}
 
-		if (formid === id || !species.cosmeticFormes || !species.cosmeticFormes.includes(formid)) {
+		if (formid === id || !species.cosmeticFormes || !species.cosmeticFormes.map(toID).includes(formid)) {
```

Comment:
Can you just do it the way I implemented it on server? That way's probably better.